nvflare/fuel/f3/drivers/driver.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) 2021-2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List
+
+from nvflare.fuel.f3.connection import Connection, ConnState
+from nvflare.fuel.f3.drivers.connector import Connector
+
+
+class ConnMonitor(ABC):
+    @abstractmethod
+    def state_change(self, connection: Connection):
+        """Driver state change notification, including new connections
+
+        Args:
+            connection: The connection that state has changed
+
+        Raises:
+            CommError: If any error happens while processing the frame
+        """
+        pass
```

Comment:
should we move this into upper layer and a separate file?
``nvflare/fuel/f3/conn_monitor.py``

Reply:
Let's talk about this later. I was asked to merge this into the driver file because we had too many small files, Java-syndrome :)